src/pm/hydra/configure.ac

```diff
@@ -142,25 +130,17 @@ AC_SUBST([mpl_srcdir])
 AC_SUBST(mpl_dist_srcdir)
 mpl_includedir=""
 AC_SUBST([mpl_includedir])
-mpl_libdir=""
-AC_SUBST([mpl_libdir])
 mpl_lib=""
 AC_SUBST([mpl_lib])
-if test "$with_mpl_prefix" = "embedded" ; then
+m4_define([mpl_embedded_dir],[mpl])
+PAC_CHECK_HEADER_LIB_EXPLICIT([mpl],[mplconfig.h],[$MPLLIBNAME],[mpl_str_get_int_arg])
+if test "$with_mpl" = "embedded" ; then
     mpl_srcdir="mpl"
     mpl_dist_srcdir="mpl"
     mpl_lib="mpl/lib${MPLLIBNAME}.la"
     mpl_subdir_args="--disable-versioning --enable-embedded"
     PAC_CONFIG_SUBDIR_ARGS([mpl],[$mpl_subdir_args],[],[AC_MSG_ERROR(MPL configure failed)])
```

Comment:
based on this comment: https://github.com/pmodels/mpich/pull/4819/commits/dc6099ec1ef90a4bcdc5ceae0b8221593c8e5337#r568008043

should this be
```
PAC_CONFIG_SUBDIR_ARGS([mpl_embedded_dir],[$mpl_subdir_args],[],[AC_MSG_ERROR(MPL configure failed)])
```

Reply:
Hmm, I tried replace `mpl` with `mpl_embedded_dir`, but the result is very difficult to read since we can't tell what happens with the `m4` variable replacement. I'll modify `hwloc` usages below instead, so the m4 variable of `xxx_embedded_dir` is only used for `m4`-logic.